test/e2e/apps/daemon_set.go

```diff
@@ -746,3 +780,189 @@ func waitFailedDaemonPodDeleted(c clientset.Interface, pod *v1.Pod) func() (bool
 		return false, nil
 	}
 }
+
+// deleteDaemonSetAndOrphanPods deletes the given DaemonSet and orphans all its dependents.
+// It also checks that all dependents are orphaned, and the DaemonSet is deleted.
+func deleteDaemonSetAndOrphanPods(
+	dsClient appstyped.DaemonSetInterface,
+	podClient corev1typed.PodInterface,
+	controllerRevisionClient appstyped.ControllerRevisionInterface,
+	ds *apps.DaemonSet) error {
+
+	ds, err := dsClient.Get(ds.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get daemonset %q: %v", ds.Name, err)
+	}
+
+	deletePropagationOrphanPolicy := metav1.DeletePropagationOrphan
+	deleteOptions := &metav1.DeleteOptions{
+		PropagationPolicy: &deletePropagationOrphanPolicy,
+		Preconditions:     metav1.NewUIDPreconditions(string(ds.UID)),
+	}
+	if err = dsClient.Delete(ds.Name, deleteOptions); err != nil {
+		return fmt.Errorf("failed to delete daemonset %q: %v", ds.Name, err)
+	}
+
+	if err = checkDaemonSetDeleted(dsClient, ds.Name); err != nil {
+		return err
+	}
+
+	if err = checkDaemonSetPodsOrphaned(podClient, ds.Name); err != nil {
+		return err
+	}
+
+	if err = checkDaemonSetHistoriesOrphaned(controllerRevisionClient, ds.Name); err != nil {
+		return err
+	}
+
```

Comment:
Please keep the check of len(orphans) > 0. The test would pass if the orphans are deleted, but it should fail in that case.